packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -172,7 +176,22 @@ export async function pnpmSyncPrepareAsync(options: IPnpmSyncPrepareOptions): Pr
 
     // if .pnpm-sync.json already exists, read it first
     if (fs.existsSync(pnpmSyncJsonPath)) {
-      pnpmSyncJsonFile = JSON.parse(fs.readFileSync(pnpmSyncJsonPath).toString());
+      const existingPnpmSyncJsonFile: IPnpmSyncJson = JSON.parse(
+        fs.readFileSync(pnpmSyncJsonPath).toString()
+      );
+      if (existingPnpmSyncJsonFile?.version === pnpmSyncJsonVersion) {
+        pnpmSyncJsonFile = existingPnpmSyncJsonFile;
+      } else {
+        logMessageCallback({
+          message: `The .pnpm-sync.json in ${pnpmSyncJsonFolder} is out of date; pnpm-sync will regenerate it.`,
+          messageKind: LogMessageKind.VERBOSE,
+          details: {
+            messageIdentifier: LogMessageIdentifier.PREPARE_WRITING_FILE,
```

Comment:
Make a new id, e.g. `PREPARE_REPLACING_FILE`

Consumers like Rush can't customize these messages if the `LogMessageIdentifier` doesn't uniquely determine the message/situation.